src/QirRuntime/test/Standalone/RandomBit/compiler/QirCore.qs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Quantum.Core{
+
+    @Attribute()
+    newtype Attribute = Unit;
+
+    @Attribute()
+    newtype Inline = Unit;
+
+    @Attribute()
+    newtype EntryPoint = Unit;
+
+    function Length<'T> (array : 'T[]) : Int { body intrinsic; }
+
+    function RangeStart (range : Range) : Int { body intrinsic; }
+
+    function RangeStep (range : Range) : Int { body intrinsic; }
+
+    function RangeEnd (range : Range) : Int { body intrinsic; }
+
+    function RangeReverse (range : Range) : Range { body intrinsic; }
+}
+
+namespace Microsoft.Quantum.Targeting {
+
+    @Attribute()
+    newtype TargetInstruction = String;
+}
```

Comment:
Are you sure you need _all these_? 
My understanding was that the `.qs` files inside of this `compiler` directory define _those Q# entities only_ (functions, operations, etc.) that your `.qs` test (e.g. `random-bit.qs`) _uses_, but those are not implemented in the compiler's libraries. I see that your .qs test uses H() and M(), so likely you only need the H() and M() definitions in `compiler/QirTarget.qs`.

Reply:
They are not needed anymore. I used them in the past to generate QIR (before the change to produce them using a Q# project was merged to `main`. I removed them in the latest iteration of changes.